CONTRIBUTING.md

```diff
@@ -175,3 +172,57 @@ $ pytest -vvx
 ### Ruby
 
 The Ruby libraries are bindings around the GLib libraries.
+
+## Opening a Pull Request
+
+Before opening a pull request, please run the static checks, which are
+enforced via [`pre-commit`](https://pre-commit.com/).  This will run
+linters, formatters, and other analysis.  For example:
+
+```shell
+$ pip install pre-commit
+# Set up hooks
+$ pre-commit install
+# Run manually
+$ pre-commit run
+Check Xml............................................(no files to check)Skipped
+Check Yaml...........................................(no files to check)Skipped
+Fix End of Files.....................................(no files to check)Skipped
+Trim Trailing Whitespace.............................(no files to check)Skipped
+clang-format.........................................(no files to check)Skipped
+cmake-format.........................................(no files to check)Skipped
+cpplint..............................................(no files to check)Skipped
+Google Java Formatter................................(no files to check)Skipped
+black................................................(no files to check)Skipped
+flake8...............................................(no files to check)Skipped
+isort................................................(no files to check)Skipped
+# Hooks automatically run on commit
+$ git commit
```

Comment:
we should probably add `go fmt` and a go linter to the pre-commit scripts

Reply:
What would be a good hook to use?

Reply:
After a quick bit of searching:

repo: https://github.com/golangci/golangci-lint
id: golangci-lint

and maybe: 

repo: https://github.com/syntaqx/git-hooks
id: go-fmt

I think? Haven't used `pre-commit` before but i think these would be good.